### PR TITLE
Fixes advanced buildmode not having a picker tool

### LINKED ISSUE #40445: Alt-click to copy obj/turf in buildmode adv. disappeared
image
`(no alt click = copy obj/turf)`
A very useful buildmode function just vanished, is this intentional? If so, ~~reeeeeeeeeeeeeeeeeeeeeeee~~

### PR DESCRIPTION
**Pull request by MrDoomBringer**
For some reason, when #40105 was merged, they neglected to implement the changes in #37337, meaning the picker tool we added was lost. This re-adds it.

:cl: MrDoomBringer
fix: Fixes advanced buildmode not having the picker
/:cl:

Fixes #40445

### CODE REVIEW

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/buildmode/submodes/advanced.dm [hunk 1]
```diff
@@ -32,7 +33,12 @@
 	var/list/pa = params2list(params)
 	var/left_click = pa.Find("left")
 	var/right_click = pa.Find("right")
-	if(left_click)
+	var/alt_click = pa.Find("alt")
+
+	if(left_click && alt_click)
+		objholder = object.type
```
[the remark this answers is not in the document]
Comment (MrDoomBringer) [reply code/modules/buildmode/submodes/advanced.dm]: uuuh are you sure? We didn't validate path before #40105. We dont really need to validate path, as this is just setting the objholder's type. The path validation happens later, once we actually try to build.

[the remark this answers is not in the document]
Comment (AnturK) [reply code/modules/buildmode/submodes/advanced.dm]: Yes, i'm sure. The fact it wasn't correct before is not really important here. If you want you can add additional check before actual object creation, but it's more about clicking a area/lighting/whatever by mistake than malicious admins.

[the remark this answers is not in the document]
Comment (MrDoomBringer) [reply code/modules/buildmode/submodes/advanced.dm]: @AnturK 
image
Cant object only be of type obj, not area?

[the remark this answers is not in the document]
Comment (AnturK) [reply code/modules/buildmode/submodes/advanced.dm]: Yep that will be good enough.

[the remark this answers is not in the document]
Comment (MrDoomBringer) [reply code/modules/buildmode/submodes/advanced.dm]: aight i went ahead and just added this to helpers cus why not

File: code/__DEFINES/is_helpers.dm [hunk 2]
```diff
@@ -8,6 +8,8 @@
 
 #define isatom(A) (isloc(A))
 
+#define istom(A) (istype(A, /turf) || istype(A, /obj) || istype(A, /mob))
```
Comment (AnturK) [inline code/__DEFINES/is_helpers.dm]: The name is incorrect. And this really shouldn't be a define it's buildmode specific restriction.

**Comment by Crazylemon64** [comment]
Oops, my bad

File: code/__DEFINES/is_helpers.dm (hunk 2 above)
Comment (MrDoomBringer) [reply code/__DEFINES/is_helpers.dm]: i mean i feel like defining things that players can see and interact with should be useful, but ok.
also nah fam tom like atom but without the area

File: code/modules/buildmode/submodes/advanced.dm [hunk 3]
```diff
@@ -32,7 +33,15 @@
 	var/list/pa = params2list(params)
 	var/left_click = pa.Find("left")
 	var/right_click = pa.Find("right")
-	if(left_click)
+	var/alt_click = pa.Find("alt")
+
+	if(left_click && alt_click)
+		if ((istype(object, /turf) || istype(object, /obj) || istype(object, /mob))
```
Comment (AnturK) [inline code/modules/buildmode/submodes/advanced.dm]: Drop the needless ()'s and it's good to go.

Comment (MrDoomBringer) [reply code/modules/buildmode/submodes/advanced.dm]: imagine a future where i had the mental abilities to make a 2-line change witout a billion reviews im gonna die

File: code/modules/buildmode/submodes/advanced.dm [hunk 4]
```diff
@@ -32,7 +33,15 @@
 	var/list/pa = params2list(params)
 	var/left_click = pa.Find("left")
 	var/right_click = pa.Find("right")
-	if(left_click)
+	var/alt_click = pa.Find("alt")
+
+	if(left_click && alt_click)
+		if (istype(object, /turf || istype(object, /obj) || istype(object, /mob))
```
Comment (AnturK) [inline code/modules/buildmode/submodes/advanced.dm]: missed one )

**Pull request merged**